src/Compilers/Core/Portable/CommandLine/CommonCompiler.ExistingReferencesResolver.cs

```diff
@@ -59,7 +57,8 @@ private static AssemblyIdentity TryGetIdentity(MetadataReference metadataReferen
 
                 try
                 {
-                    return ((AssemblyMetadata)peReference.GetMetadataNoCopy()).GetAssembly().Identity;
+                    PEAssembly assembly = ((AssemblyMetadata)peReference.GetMetadataNoCopy()).GetAssembly()!;
```

Comment:
>! [](start = 107, length = 1)

nit: if we're going to split the line and use a temp anyways, what's the purpose of using a suppression versus assertion?

Reply:
My rational for adding the temp here is that I just couldn't mentally grok the line when I added the `!` inline. Too many parens for me to know at a glance what was going on. Hence I wasn't trying to change behavior here, just make the code clearer. 

I can add an assert though if you think it's a better approach.